crates/oidc-client/src/requests/authorization_code.rs

```diff
@@ -74,9 +77,39 @@ pub struct AuthorizationRequestData<'a> {
     ///
     /// It must be one of the redirect URIs provided during registration.
     pub redirect_uri: &'a Url,
+}
+
+/// Extra parameters to influence the authorization flow.
+#[derive(Debug, Default, Clone)]
+pub struct AuthorizationRequestExtraParameters {
```

Comment:
Sure we can do that. I was hesitating between the two of them

Reply:
I decided to pull `code_challenge_methods_supported` out of the struct because it's the only type that we only need borrowed. It doesn't make sense for it to be in the struct as owned, or to have a struct with only borrowed types when we need almost all of them owned.